categorizing of specializations (#105)
CaptainSifff/paper_teaching-learning-RSE: competencies.md

But not all RSEs are created equal, they specialize in different areas,
some of which we want to present below. Many of the specializations may overlap,
so the same RSE might for example work on data management and on Open Science.
We figure many RSEs to have more than one of the specializations mentioned below.
We categorize them into those, that can be viewed as a specialization within RSE-specific topics,

> while other RSEs might epxand their skill set and profession to areas, that are not typical for a RSE.

Reviewer: small typo. epxand -> expand

Author: ```suggestion
while other RSEs might expand their skill set and profession to areas, that are not typical for a RSE.
```

[BIO Excel framework](https://competency.ebi.ac.uk/framework/bioexcel/3.0/carreer-profiles)

## RSE specializations
What we have defined above are intended to be base skills that a RSE
irrespective of domain, place, and time should know about.

> But not all RSEs are created equal, they specialize in different areas,

Reviewer 2: ```suggestion
But beyond these core skills, RSEs specialize in different areas,
```

I do not like "created equal" here.

The research environment is constantly changing and this can also apply to the software requirements.
If software is not properly maintained, it becomes increasingly difficult to install and use. 
At some point, the software is no longer available and cannot be used to reproduce results.
While ensuring maintenance and sustainability of research software is of huge importance to the communities that build and use it,
a particular challenge is that it is often very difficult to obtain ongoing research funding for software maintenance tasks.
As a result, when a project that developed or extended a piece of software finishes, support for the software fades as team members move on to other research,
academic or RSE roles, or become busy with other funded work.
While this is not a core concern of this paper, we wanted to highlight this important issue that is frequently faced when working with software in the research community.
With regard to which additional competency is required,
these are people having experience with ancient software stacks that are not anymore part of the general curricula (think of COBOL and FORTRAN).

> FIXME: I think it would be nice if we could move each of these optional competencies to a different specialization.

Reviewer 3: I agree, this should be shortened. What you have moved out, is more political and could fit to the call-for-action part.

> #### Optional RSE competencies -> Maintenance RSEs 
> Oftentimes, a significant amount of effort in (research) software development needs to be spent on maintenance to ensure that software remains useful for researchers now and in the future.
> The research environment is constantly changing and this can also apply to the software requirements.
> If software is not properly maintained, it becomes increasingly difficult to install and use. 
> At some point, the software is no longer available and cannot be used to reproduce results.
> While ensuring maintenance and sustainability of research software is of huge importance to the communities that build and use it,
> a particular challenge is that it is often very difficult to obtain ongoing research funding for software maintenance tasks.
> As a result, when a project that developed or extended a piece of software finishes, support for the software fades as team members move on to other research,
> academic or RSE roles, or become busy with other funded work.
> While this is not a core concern of this paper, we wanted to highlight this important issue that is frequently faced when working with software in the research community.
> With regard to which additional competency is required,
> these are people having experience with ancient software stacks that are not anymore part of the general curricula (think of COBOL and FORTRAN).
> 
> FIXME: I think it would be nice if we could move each of these optional competencies to a different specialization.

Author: Great, I will move it there.

```suggestion
#### Maintenance RSEs 
In a constantly changing environment, a significant amount of effort in (research) software development
needs to be spent on maintenance to ensure that software remains useful or even installable.
This can prevent reproducibility.
With regard to which additional competency is required,
these are people having experience with ancient software stacks that are not part of the general curricula anymore (think of COBOL and FORTRAN).
Another skillset required is (automated) testing.
```

### Specializations outside the core RSE competencies

> #### Legal-RSE

Author: Yeah, I think many of the things are debatable. I put this one here because
- writing/debugging/developing/... software has little to do with licenses
- every researcher should have a basic understanding of copyright, good sc. practice and licenses. RSEs might have a little bit more knowledge on the software-centered licenses and mixing them.
I think of it as something that you have to have in mind additionally to your actual (R) Software Engineering skills and techniques.

Concerning the text, I read it with all that in mind, I have some more remarks myself:

> They are the go-to person if people have a question about licensing, mixing and matching of software as well as patenting or copyright.

patenting? NO! Go see a patent lawyer.
Also very unusual in research at university.

> The publication of research software also requires knowledge about particular laws or regulatory frameworks concerning data protection, like the GDPR within the EU [@GDPR].

I think this is the conjunction of OpenScience RSE / data-focused RSE and legal RSE.

> Familiarity with legal aspects of cybersecurity and export control in science and research (see [@ExportControl] for Germany) complete the competencies of those RSEs.

Should we introduce a 'security RSE'? I guess the 'legal aspects of cybersecurity' are something different than actual security.
Cybersecurity also belongs to the web-development RSE.

> #### Legal-RSE

Reviewer 3: I'm not sure about the patenting thing... At least my university has an entire department for that:
https://www.uni-wuerzburg.de/sft/

> #### Legal-RSE

Author: Another tought on this: Legal is not just patents.

I am on a workshop these days. In one break we discussed the legal implications of providing CI capacities on the university cluster. The security and legal concerns are extreme. That is another case where you go to your legal department.

> #### Legal-RSE

Author: Maybe we rename legal to license or intelectual property? The name confuses me.
A first idea on a classification:  
core competency
- licenses (only basic knowledge)
- (privacy: awarness, that you are not allowed to publish certain data. But that is valid for any researcher)?

not core competency:
- patents
- anything else

> #### Legal-RSE

Reviewer 3: As you've found out, you need someone with an awareness of these implications, in a group.

> #### Legal-RSE

Author: I put it back in the core competencies and modified the description a bit. Hope you like it!

https://github.com/CaptainSifff/paper_teaching-learning-RSE/blob/e81871976504d672bdaa000f4b5da27a051ee53b/competencies.md?plain=1#L427-L433

> #### Legal-RSE

Author: + Revisit Software distribution in Research Skills to properly include current `Legal` in Software Engineering.
+ Remove current `Legal` in SE

_Originally posted by @hvwaldow in https://github.com/CaptainSifff/paper_teaching-learning-RSE/pull/115#pullrequestreview-1628708115_

e.g. setup of virtual machines on hypervisors or the planning and setup of compute server clusters for GPU based machine learning.
As an interface between the researchers and the infrastructure, they take care of user management, access permissions, and configuration of required services, for example.

#### Legal-RSE
With the prevalence of software, we foresee the need for RSEs that specialize in legal questions around software.

> They are the first go-to person if people have a question about licensing, mixing and matching of software as well as patenting or copyright.

Reviewer 2: ```suggestion
They are the go-to person for questions about licensing, mixing and matching of software as well as patenting or copyright.
```

Further specialization in legal questions can be in the area of publication of research software,
as this also requires knowledge about particular laws or regulatory frameworks concerning data protection, like the GDPR within the EU [@GDPR].
Another area are legal aspects of cybersecurity and export control in science and research (see [@ExportControl] for Germany).
They are able to communicate and collaborate with lawyers.

#### Maintenance RSEs 
The constantly evolving software environment can hinder or prevent reproducibility.
In this changing environment, a significant amount of effort in (research) software development
needs to be spent on maintenance to ensure that software remains useful or even installable.
With regard to which additional competency is required,
these are people having experience with ancient software stacks that are not part of the general curricula anymore (think of COBOL and FORTRAN).
Another skillset required is (automated) testing.

### Specializations outside the core RSE competencies

> #### Web-Development RSE

Reviewer 2: Suggest to reorder these specialisations alphabetically.

Author: for [ #ac988](https://github.com/CaptainSifff/paper_teaching-learning-RSE/pull/105/commits/ac9885c1cdf8759916612e56f8de5602339b9895) I ordered them according to their focus. That feels more natural for me. Added comments in markdown indicating that.
However, I would fine with alphabetically.
